archunit/src/main/java/com/tngtech/archunit/core/domain/JavaClass.java

```diff
@@ -1622,6 +1634,36 @@ private boolean belongsTo(JavaClass input, Class<?> clazz) {
             }
         }
 
+        private static class DoesNotBelongToAnyOfPredicate extends DescribedPredicate<JavaClass> {
+            private final Class<?>[] classes;
+
+            DoesNotBelongToAnyOfPredicate(Class<?>... classes) {
+                super("does not belong to any of " + JavaClass.namesOf(classes));
+                this.classes = classes;
+            }
+
+            @Override
+            public boolean apply(JavaClass input) {
+                for (Class<?> clazz : classes) {
+                    if (belongsTo(input, clazz)) {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
```

Comment:
The logic for this is exactly the same as in `BelongToAnyOfPredicate`, but with `false` and `true` inverted:

```java
            @Override
            public boolean apply(JavaClass input) {
                for (Class<?> clazz : classes) {
                    if (belongsTo(input, clazz)) {
                        return true;
                    }
                }
                return false;
            }
```